network-config-analyzer/IstioSidecarYamlParser.py

```diff
@@ -0,0 +1,195 @@
+            host_peers |= self._get_peers_from_host_namespace(namespace)
+            self._validate_dns_name_pattern(dns_name)
+            if host_peers:  # if there are services in the specified namespace
+                if '*' not in dns_name:  # * means all services in namespace, all already in host_peers
+                    host_peers &= self._get_peers_from_host_dns_name(dns_name)
+            res_peers |= host_peers
+
+        return IstioSidecarRule(res_peers)
+
+    def _append_sidecar_into_relevant_list(self, sidecar_full_name, selected_peers, is_default_sidecar):
+        """
+        save the current (self) sidecar in the relevant list.
+        if the sidecar is selector less, then appends it to the current namespace's list
+        otherwise, appends it to the lists of the selected peers in this sidecar
+        :param str sidecar_full_name: the sidecar name in this form: <namespace>/<name>
+        :param list selected_peers: the peers selected by the workloadSelector of the sidecar
+        :param bool is_default_sidecar: indicates if the sidecar is selector-less (default in its namespace)
+        a warning message will be printed if current sidecar is not the first in its relevant list,
+        indicating that this sidecar will be ignored in the sidecar's connections
+        """
+        if is_default_sidecar:
+            if self.namespace.ordered_default_sidecars:  # this sidecar is not first one
+                self.warning(f'Namespace "{str(self.namespace)}" already has a Sidecar configuration '
+                             f'without any workloadSelector. '
+                             f'Connections in sidecar: "{sidecar_full_name}" will be ignored')
+            self.namespace.ordered_default_sidecars.append(sidecar_full_name)
+            return
+
+        for peer in selected_peers:
+            if peer.ordered_specific_sidecars:  # this sidecar is not first one
+                self.warning(f'Peer "{peer.full_name()}" already has a Sidecar configuration selecting it.'
+                             f'Sidecar: "{sidecar_full_name}" will not be considered as connections for this workload')
+            peer.ordered_specific_sidecars.append(sidecar_full_name)
+
+    def parse_policy(self):
+        """
+        Parses the input object to create a IstioSidecar object
+        :return: a IstioSidecar object with proper PeerSets
+        :rtype: IstioSidecar
+        """
+        policy_name = self.parse_generic_istio_policy_fields('Sidecar', 'networking.istio.io/v1beta1')
+        if policy_name is None:
+            return None  # not relevant to build this policy
+
+        res_policy = IstioSidecar(policy_name, self.namespace)
+        res_policy.policy_kind = NetworkPolicy.PolicyType.IstioSidecar
+
+        if 'spec' not in self.policy or self.policy['spec'] is None:
+            self.warning('spec is missing or null in Sidecar ' + res_policy.full_name())
```

Comment:
are we sure this is not a syntax error?
Better try on a live cluster

Reply:
true, it is a syntax error.
trying to apply on live cluster both authorization-policy and a sidecar without spec
got a server error for the sidecar wo the spec, while the auth-policy was created